kolibri/core/assets/src/vue/core-modal/index.vue

```diff
@@ -109,6 +102,9 @@
       closeModal() {
         this.$dispatch('close');
       },
+      submitModal() {
+        this.$dispatch('submit');
+      },
```

Comment:
I see what you're saying about listening for user changes, but the way I understood it, the modal was to be as generic as possible. It doesn't impose any opinions about whether a user is supposed to submit or cancel anything (since, presumably, the modal doesn't always have to be a form).

Ir being generic, the local and the content that populate it are, in my mind, separste. Modal is listening for close because that's how it knows when the close button (that all modals will have) is clicked. It emits open and close so that the content that's going to populate the modal (the parent) has events to respond to, like clearing data when opened or running some method when closed.

I didn't know about attached/detached. They seem to be good for the stuff happening internally, but it doesn't broadcast so the parent wouldn't be able to know when the model was closed via the x button rather than the parent's close button so the stuff described above wouldn't be possible.

I'm still out, but I can read more into this stuff once I get back. Typed this out on my phone haha. If it needs to be merged I don't have any objections, but I agree that we need to discuss and come to an agreement.